src/update_engine/utils.cc

```diff
@@ -502,16 +502,19 @@ const std::string BootDevice() {
   return boot_path;
 }
 
-const string BootKernelDevice(const std::string& boot_device) {
-  // Currntly this assumes the last digit of the boot device is
-  // 3, 5, or 7, and changes it to 2, 4, or 6, respectively, to
-  // get the kernel device.
+const string BootKernelName(const std::string& boot_device) {
```

Comment:
Actually, ret will never be returned based on the boot_device value so the variable `ret` could be dropped entirely.